darkmoon/api/v1/metadata/views.py

```diff
@@ -493,19 +478,26 @@ async def hash_comparison(
             ]
             inputHashes = [md5Hash, sha1Hash, sha256Hash, sha512Hash]
             if dbHashes != inputHashes:
-                await susCollection.insert_one(obj.dict())
+                insert_result = await susCollection.insert_one(obj)
+                inserted_id = str(insert_result.inserted_id)
+                obj["_id"] = inserted_id
                 response.status_code = status.HTTP_406_NOT_ACCEPTABLE
+                data: list[MetadataEntity] = []
+                if "header_info" in obj:
+                    data.append(EXEMetadataEntity.parse_obj(obj))
+                else:
+                    data.append(DocMetadataEntity.parse_obj(obj))
                 return UploadListMetadataEntityResponse(
                     message="Bad hashes. Put in suspicious collection.",
-                    data=[obj],
+                    data=data,
                 )
         search_query = {
             "$or": [
                 {
                     "name": inputFileName,
                 },
                 {
-                    "file_type": inputFileType,
+                    "type": inputFileType,
```

Comment:
What is the value of searching for a name *OR* a filetype? Wouldn't the filetype have a ton of matches and cause false-positives in the suspicious files?